### ISSUE TITLE
Replace Last Param in `DirectiveHelper.compileFn()` with Options Literal

### ISSUE THREAD

**Issue by jdalt** [jdalt/angular-test-tool-box#4]
The last param `DirectiveHelper.compileFn()` is confusing and it's use is not obvious. Replace it with an options object literal that can alter the compile mechanism in predefined ways.

**Comment by ScottTrenda** [comment]
This version of `compileFn` should be backwards compatible:
```
  function compileFn(tmpl, options) {

    options = (typeof options == 'boolean') ? { flushRequests: options } : options || {}

    return function compile(scopeParams) {
      scopeParams = scopeParams || {}
      var scope = $rootScope.$new()

      angular.extend(scope, scopeParams)

      var el = $compile(tmpl)(scope)
      scope.$digest()

      if (options.flushRequests || options.flushRequests === undefined) $httpBackend.flush()

      return angular.extend({ scope: scope }, DomHelper(el))
    }

  }
```

**Comment by ScottTrenda** [comment]
Actually, let's kill another few birds with this stone and do it this way:
```
function compileFn(tmpl, options) {

  options = (typeof options == 'boolean') ? { flushRequests: options } : options || {}

  return function compile(scopeParams, overrideOptions) {
    scopeParams = scopeParams || {}
    var scope = $rootScope.$new()

    angular.extend(scope, scopeParams)
    var callOptions = angular.extend({}, options, overrideOptions)

    var cloneAttachFn
    if (callOptions.attach) cloneAttachFn = function(clone) { angular.element('body').append(clone) }

    var el = $compile(tmpl)(scope, cloneAttachFn)
    scope.$digest()

    if (callOptions.flushRequests || callOptions.flushRequests === undefined) $httpBackend.flush()

    return angular.extend({ scope: scope }, DomHelper(el))
  }

}
```
Allowing options to be passed on the `compile()` call will solve the annoying case where we want the requests to be flushed on compile for every test except one in a suite. We'll be able to pass `{ flushRequests: false }` for only that call.

The `attach` option is there because we might as well, while we're updating this. I'd love to also automate detaching the element at the end of the current test run if `attach` is passed, but I don't know if or how we can add an `afterEach` from within an `it` block. The tests using `{ attach: true }` will have to manually detach their own element afterward, until we can figure that out.

**Comment by ScottTrenda** [comment]
Addressed in #9
